engines/dfc_provider/spec/requests/supplied_products_spec.rb

```diff
@@ -1,91 +1,112 @@
 # frozen_string_literal: true
 
+require "swagger_helper"
 require DfcProvider::Engine.root.join("spec/spec_helper")
 
-describe "SuppliedProducts", type: :request do
+describe "SuppliedProducts", type: :request, swagger_doc: "dfc-v1.7/swagger.yaml" do
   let!(:user) { create(:oidc_user) }
   let!(:enterprise) { create(:distributor_enterprise, owner: user) }
   let!(:product) { create(:simple_product, supplier: enterprise ) }
   let!(:variant) { product.variants.first }
 
-  describe :create do
-    let(:endpoint) do
-      enterprise_supplied_products_path(enterprise_id: enterprise.id)
-    end
-    let(:supplied_product) do
-      SuppliedProductBuilder.supplied_product(new_variant)
-    end
-    let(:new_variant) do
-      # We need an id to generate a URL as semantic id when exporting.
-      build(:variant, id: 0, name: "Apple", unit_value: 3)
-    end
+  before { login_as user }
 
-    it "flags a bad request" do
-      post endpoint, headers: auth_header(user.uid)
+  path "/api/dfc-v1.7/enterprises/{enterprise_id}/supplied_products" do
+    parameter name: :enterprise_id, in: :path, type: :string
 
-      expect(response).to have_http_status :bad_request
-    end
+    let(:enterprise_id) { enterprise.id }
+
+    post "Create SuppliedProduct" do
+      consumes "application/json"
+      produces "application/json"
 
-    it "creates a variant" do
-      request_body = DfcLoader.connector.export(supplied_product)
+      # This parameter is required but I want to write a spec which doesn't
+      # supply it. I couldn't do it with rswag when requiring it.
```

Comment:
Interesting, I suppose this feature needs to be added to rswag one day..

Or ideally, I wonder if  rswag could automatically add a test for a `400` when `required:true`? Either way, you'd want to be able to override it.